src/databricks/labs/ucx/azure/resources.py

```diff
@@ -76,6 +79,7 @@ class Principal:
     # Need this directory_id/tenant_id when create UC storage credentials using service principal
     # it will be None if type is managed identity
     directory_id: str | None = None
+    secret: str | None = None
```

Comment:
```suggestion
    secret: str
```
In your code, secret is not none. Reflect it on the dataclass.

Reply:
Principal Data class is used by principal_prefix_access functionality which doesn't use secret hence I have created it as optional